spirv_msl.cpp

```diff
@@ -1365,8 +1371,11 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 		// Mark that this shader reads from this image
 		uint32_t img_id = ops[2];
 		auto *p_var = maybe_get_backing_variable(img_id);
-		if (p_var)
+		if (p_var && has_decoration(p_var->self, DecorationNonReadable))
```

Comment:
Is there something similar for NonWriteable?

Reply:
yes, the code was already there